parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java

```diff
@@ -892,8 +892,9 @@ private void doEnqueue(long scheduledExecutionTime) {
 
             // Update `ts` for the event since it's happening right now.
             Calendar now = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
-            Map<String, Object> data = (Map<String, Object>) event.get("data");
+            Map<String, Object> data = new HashMap<>((Map<String, Object>) event.get("data"));
```

Comment:
1. Suggestion (💡): How would you feel about suppressing the `unchecked` warning on `(Map<String, Object>)` to make it explicit that we know what we are doing here?
2. Suggestion (💡): How would you feel about using `requireNonNull(...)` to make it explicit that we expect `event.get("data")` to never be null?
3. Minor (🔍): Not related to your change, but since you correctly used `new HashMap<>` instead of `new HashMap`, consider changing the above such construction from `new HashMap(this.baseEvent)` to `new HashMap<>(this.baseEvent)`.

PS: With the above suggestions and minor improvements I am hoping to end-up with a warning-free `doEnqueue(...)` method. Feel free to ignore or apply any. 💯

Reply:
Thank you for the review @ParaskP7 ! 

> 1. Suggestion (💡): How would you feel about suppressing the unchecked warning on (Map<String, Object>) to make it explicit that we know what we are doing here?

Sure 👍 

> 2. Suggestion (💡): How would you feel about using requireNonNull(...) to make it explicit that we expect event.get("data") to never be null?

I've decided to not force non-nullable `event.get("data")` to not crash users applications. It shouldn't ever happen, but I thought it's better safe than sorry in this case (especially for libraries). WDYT?

> 3. Minor (🔍): Not related to your change, but since you correctly used new HashMap<> instead of new HashMap, consider changing the above such construction from new HashMap(this.baseEvent) to new HashMap<>(this.baseEvent).

Sure! Now `doEnqueue` method is warning free. The PR is ready for the second round!

Reply:
> this peace of code was there for years and already tested (by our users)

On the second thought, actually that's a good point 👍 . I've brought back the fail-fast behavior. I used `assert` instead of `requireNonNull` as it's not Kotlin (yet) 😄. https://github.com/Parsely/parsely-android/pull/62/commits/bc62a92deab5452e0085ca419fa2e7b091316cc9

Reply:
Awesome, thank @wzieba ! 🚀 

> I've brought back the fail-fast behavior. I used assert instead of requireNonNull as it's not Kotlin (yet) 😄.

`assert` or whichever way you do it is totally fine by me, but just as an FYI `Objects.requireNonNull(event.get("data")` works for Java too. 😅

```java
Map<String, Object> data = new HashMap<>(
        (Map<String, Object>) Objects.requireNonNull(event.get("data"))
);
```

Feel free to merge whenever, no need to change anything! 🚀

Reply:
TIL, thanks! I merged the PR before reading the comment but I'll know this for the future 🙂